packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js

```diff
@@ -82,38 +80,45 @@ export default ({ coreSagas }) => {
       yield put(A.coinifyLoading())
       const trade = yield call(coreSagas.data.coinify.sell)
 
+      const state = yield select()
+
       if (!trade) {
         const trade = yield select(selectors.core.data.coinify.getTrade)
         const parsed = JSON.parse(trade.error)
 
         yield put(A.coinifyFailure(parsed))
         return
       }
-      const p = yield select(sendBtcSelectors.getPayment)
+      const p = path(['coinify', 'payment'], state)
       let payment = yield coreSagas.payment.btc.create(
         { payment: p.getOrElse({}), network: settings.NETWORK })
       payment = yield payment.amount(parseInt(trade.sendAmount))
 
       // QA Tool: manually set a "to" address on the payment object for testing sell
-      const qaState = yield select()
-      const qaAddress = path(['qa', 'qaSellAddress'], qaState)
-
+      const qaAddress = path(['qa', 'qaSellAddress'], state)
       if (qaAddress) {
```

Comment:
shouldn't we make such address injections dev/staging only?

Reply:
What if we want to test in a prod environment?

Reply:
Indeed, good point